packages/beeq/src/components/drawer/bq-drawer.tsx

```diff
@@ -0,0 +1,250 @@
+import { Component, Element, Event, EventEmitter, h, Host, Listen, Method, Prop, State, Watch } from '@stencil/core';
+import { enter, leave } from 'el-transition';
+
+import { TDrawerPlacement } from './bq-drawer.types';
+import { hasSlotContent } from '../../shared/utils';
+
+/**
+ * @part body - The `<main>` that holds the drawer body content
+ * @part button-close - The button that close the dialog on click
+ * @part content - The `<div>` container that holds the drawer title and body content
+ * @part wrapper - The `<div>` wrapper container inside the shadow DOM
+ * @part footer - The `<footer>` that holds footer content
+ * @part header - The `<header>` that holds the icon, title, and close button
+ * @part title - The `<div>` that holds the title content
+ */
+
+@Component({
+  tag: 'bq-drawer',
+  styleUrl: './scss/bq-drawer.scss',
+  shadow: true,
+})
+export class BqDrawer {
+  // Own Properties
+  // ====================
+
+  private drawerElem: HTMLDivElement;
+  private footerElem: HTMLElement;
+
+  // Reference to host HTML element
+  // ===================================
+
+  @Element() el!: HTMLBqDrawerElement;
+
+  // State() variables
+  // Inlined decorator, alphabetical order
+  // =======================================
+
+  @State() private hasFooter = false;
+  @State() hasIcon: boolean = false;
+
+  // Public Property API
+  // ========================
+
+  /** If true, the drawer component will be shown */
+  @Prop({ reflect: true, mutable: true }) open: boolean;
+
+  /** Defines the position of the drawer */
+  @Prop({ reflect: true, mutable: true }) placement?: TDrawerPlacement = 'left';
```

Comment:
Use `validatePropValue` to make sure placement is always `TDrawerPlacement`
```suggestion
  @Prop({ reflect: true, mutable: true }) placement: TDrawerPlacement = 'left';
```